core/src/sumeragi/mod.rs

```diff
@@ -201,25 +201,28 @@ impl SumeragiHandle {
     }
 
     /// Deposit a sumeragi network message.
-    pub fn incoming_message(&self, msg: MessagePacket) {
-        if msg.message.is_none() {
-            if let Err(error) = self.control_message_sender.try_send(ControlFlowMessage {
-                view_change_proofs: msg.view_change_proofs,
-            }) {
-                self.metrics.dropped_messages.inc();
-                error!(
-                    ?error,
-                    "This peer is faulty. \
-                     Incoming control messages have to be dropped due to low processing speed."
-                );
+    pub fn incoming_message(&self, msg: Message) {
+        match msg {
+            Message::ControlFlow(msg) => {
+                if let Err(error) = self.control_message_sender.try_send(msg) {
+                    self.metrics.dropped_messages.inc();
+                    error!(
+                        ?error,
+                        "This peer is faulty. \
+                         Incoming control messages have to be dropped due to low processing speed."
+                    );
+                }
+            }
+            Message::Block(msg) => {
```

Comment:
```suggestion
            Message::Data(msg) => {
```

how about this? I would say we have data and control flow messages